instrument/options.go

```diff
@@ -35,6 +37,7 @@ const (
 
 type options struct {
 	logger         xlog.Logger
+	zap            *zap.Logger
```

Comment:
Instead of this, would it be better to write a `xlog.Logger` implementation that wrapped `zap.Logger`?

Reply:
Long term we should straight up replace `xlog.Logger` with `zap.Logger` - for the reasons you list, and also because Zap uses a more performant `LogField` implementation (can specify which type serialiser to use, avoid allocs unless necessary).

That might be too pervasive to start tho, starting with the wrapped impl would be easier.

Reply:
Thought about this some more, exposing the `zap.Logger` type (the way you're doing it), is a better middle ground. 

Reasoning (for posterity): this way allows downstream packages (m3db/m3aggregator/etc) to get off our logger implementation without breaking other consumers. Will make the process to remove our own implementation of log easier. Wrapping the Zap impl behind our interface is throw away work in the long term.